api/v1/ptpconfig_webhook.go

```diff
@@ -136,21 +139,18 @@ func (r *PtpConfig) ValidateDelete() error {
 	return nil
 }
 
-func getInterfaces(input *ptp4lConf, mode string) (interfaces []string) {
-	modeInt := "0"
-	if mode == Master {
-		modeInt = "1"
-	}
+func getInterfaces(input *ptp4lConf, mode PtpRole) (interfaces []string) {
+
 	for index, section := range input.sections {
 		sectionName := strings.TrimSpace(strings.ReplaceAll(strings.ReplaceAll(index, "[", ""), "]", ""))
-		if strings.TrimSpace(section.options["masterOnly"]) == modeInt {
+		if strings.TrimSpace(section.options["masterOnly"]) == strconv.Itoa(int(mode)) {
```

Comment:
masterOnly is not a valid int, the validation webhook will crash?

Reply:
"mode" is of type PtpRole int, which is an int. We should be ok. Also, I did not touch the original code at all. The reason I added the get interface function here was to reuse the code in populatePtp4lConf. I see 3 options:
- add a function in ptpconfig_webhook.go without touching the original code (current PR code)
- modify the original code to make the PopulatePtp4lConf usable in test code
- duplicate the populatePtp4lConf so there is no change to the ptpconfig_webhook.go file or to the code (previous solution before suggestion from Joseph to reuse the code)
Which one should we implement? Do we need a bugzilla to edit the ptpconfig_webhook.go even if no changes to existing code?

Reply:
alright!, since  @josephdrichard has okayed to add these changes to webhook., let's keep it. No need to add bugsZilla since this code will be not be used by PTP.